__tests__/app.test.js

```diff
@@ -125,4 +126,96 @@ describe("/api/articles", () => {
         
         })
     })
+})
+
+
+describe("/api/articles/:article_id/comments", () => {
+    test("GET 200: Responds with a 200 OK status", () => {
+        return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+    })
+
+    test("GET 200: Responds with a list of the correct length", () => {
+        return request(app)
+        .get("/api/articles/1/comments")
+        .then(({body})=> {
+           const {comments} = body
+           expect(comments.length).toBe(11)
+
+        })
+    })
+
+    test("GET 200: Responds with a list of all comments from a particular article", () => {
+        return request(app)
+        .get("/api/articles/1/comments")
+        .then(({body})=> {
+           const {comments} = body
+
+           comments.forEach((comment)=>{
+            expect(comment.article_id).toBe(1)
+           })
+
+        })
+    })
+
+    test("GET 200: Response contains all of the correct object keys", () => {
+        return request(app)
+        .get("/api/articles/1/comments")
+        .then(({body})=> {
+           const {comments} = body
+
+           comments.forEach((comment)=>{
+            expect(comment).toMatchObject({
+                comment_id: expect.any(Number),
+                body: expect.any(String),
+                article_id: expect.any(Number),
+                author: expect.any(String),
+                votes: expect.any(Number),
+                created_at: expect.any(String),
+            })
+        })
+
+        })
+    })
+
+    test("GET 200: Comments are all ordered by created_at: descending", () => {
+        return request(app)
+        .get("/api/articles/1/comments")
+        .then(({body})=> {
+           const {comments} = body
+
+            expect(comments).toBeSortedBy("created_at", {descending: true})
+
+        })
+    })
+
+    test("GET 400: Returns a 400 (Bad request) when an incorrect article_id format is entered", () => {
+        return request(app)
+        .get("/api/articles/firstArticle/comments")
+        .expect(400)
+        .then(({body})=> {
+            expect(body.msg).toBe("Bad Request")
+        })
+    })
+
+    test("GET 404: Returns a 404 (Not Found) when a non-existant article_id is entered", () => {
+        return request(app)
+        .get("/api/articles/999/comments")
+        .expect(404)
+        .then(({body})=> {
+            expect(body.msg).toBe("Not Found")
+        })
+    })
+
+    // THIS TEST NEEDS TO GIVE A DIFFERENT ERROR MESSAGE BACK
```

Comment:
Actually this is fine, I wouldn't expect an error in this case as nothing has gone wrong. The correct response has been served.